File: python/tests/test_load_env.py
import os


def test_load_repo_env_sets_wandb_key(tmp_path, monkeypatch):
    monkeypatch.delenv("WANDB_API_KEY", raising=False)
    (tmp_path / ".env").write_text("WANDB_API_KEY=test-key-from-dotenv\n", encoding="utf-8")
    from load_env import load_repo_env, wandb_api_key

    path = load_repo_env(tmp_path)
    assert path == tmp_path / ".env"
    assert os.environ["WANDB_API_KEY"] == "test-key-from-dotenv"
    assert wandb_api_key(tmp_path) == "test-key-from-dotenv"

File: tests/sense/test_load_env.js
const { test } = require('node:test')
const assert = require('node:assert/strict')
const fs = require('fs')
const os = require('os')
const path = require('path')
const { loadRepoEnv } = require('../../js/sense/loadEnv')

test('loadRepoEnv reads WANDB_API_KEY from .env', () => {
  const dir = fs.mkdtempSync(path.join(os.tmpdir(), 'fly-env-'))
  fs.writeFileSync(path.join(dir, '.env'), 'WANDB_API_KEY=test-key-from-dotenv\n')
  delete process.env.WANDB_API_KEY
  const loaded = loadRepoEnv(dir)
  assert.equal(loaded, path.join(dir, '.env'))
  assert.equal(process.env.WANDB_API_KEY, 'test-key-from-dotenv')
})
